cmd/resources/gen_resources.go

```diff
@@ -0,0 +1,138 @@
+package resources
+
+import (
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type TemplateData struct {
+	Resources map[string]*ResourceData
+}
+
+type ResourceData struct {
+	Name        string
+	Description string
+	Operations  map[string]*OperationData
+}
+
+type OperationData struct {
+	Short        string
+	Long         string
+	Use          string
+	Params       []*Param
+	HTTPMethod   string
+	RequiresBody bool
+	Path         string
+}
+
+type Param struct {
+	Name        string
+	In          string
+	Description string
+	Type        string
+	Required    bool
+}
+
+func GetTemplateData(fileName string) (TemplateData, error) {
```

Comment:
this isn't actually being called anywhere but the tests yet, but I was hacking into `main.go` and passing in the file name to test it out locally. will work on actually calling this in a generate file in a follow up PR.